src/fileoverview_comment_transformer.ts

```diff
@@ -75,10 +68,58 @@ function augmentFileoverviewComments(tags: jsdoc.Tag[]) {
 export function transformFileoverviewComment(context: ts.TransformationContext):
     (sf: ts.SourceFile) => ts.SourceFile {
   return (sf: ts.SourceFile) => {
-    let comments: ts.SynthesizedComment[] = [];
-    // Use trailing comments because that's what transformer_util.ts creates (i.e. by convention).
-    if (sf.statements.length && sf.statements[0].kind === ts.SyntaxKind.NotEmittedStatement) {
-      comments = ts.getSyntheticTrailingComments(sf.statements[0]) || [];
+    const text = sf.getFullText();
+
+    let fileComments: ts.SynthesizedComment[] = [];
+    const firstStatement = sf.statements.length && sf.statements[0] || null;
+
+    const originalComments = ts.getLeadingCommentRanges(text, 0) || [];
+    if (!firstStatement) {
+      // In an empty source file, all comments are file-level comments.
+      fileComments = synthesizeCommentRanges(sf, originalComments);
+    } else {
+      // Search for the last comment split from the file with a \n\n. All comments before that are
+      // considered fileoverview comments, all comments after that belong to the next statement(s).
+      // If none found, comments remains empty, and the code below will insert a new fileoverview
+      // comment.
+      for (let i = originalComments.length - 1; i >= 0; i--) {
+        const end = originalComments[i].end;
+        if (!text.substring(end).startsWith('\n\n')) continue;
+        // This comment is separated from the source file with a double break, marking it (and any
+        // preceding comments) as a file-level comment. Split them off and attach them onto a
+        // NotEmittedStatement, so that they do not get lost later on.
+        const synthesizedComments = jsdoc.synthesizeLeadingComments(firstStatement);
+        const notEmitted = ts.createNotEmittedStatement(sf);
+        // Modify the comments on the firstStatement in place by removing the file-level comments.
+        fileComments = synthesizedComments.splice(0, i + 1);
+        // Move the fileComments onto notEmitted.
+        ts.setSyntheticLeadingComments(notEmitted, fileComments);
+        // TODO(martinprobst): consider checking here whether any of the trailing comments contains
+        // an @fileoverview etc, and reporting an error if so.
+        sf = updateSourceFileNode(
+            sf, ts.createNodeArray([notEmitted, firstStatement, ...sf.statements.slice(1)]));
+        break;
+      }
+
+      // Now walk every top level statement and escape/drop any @fileoverview comments found.
+      // Closure ignores all @fileoverview comments but the last, so tsickle must make sure not to
+      // emit duplicated ones.
+      // TODO(martinprobst): make this an error instead of silently dropping the comment.
```

Comment:
It's easy to find the failures, it's just a global presubmit. I just wanted to avoid front-loading more global changes into this PR, and keep the difference to the current behaviour of tsickle small.